R-package/NAMESPACE

```diff
@@ -32,6 +32,10 @@ export(slice)
 export(xgb.DMatrix)
 export(xgb.DMatrix.hasinfo)
 export(xgb.DMatrix.save)
+export(xgb.DataIter)
+export(xgb.ExternalDMatrix)
+export(xgb.ProxyDMatrix)
```

Comment:
I'm not sure if we need to export the proxy DMatrix, users are unlikely to interact with it.